File: src/main/java/com/app/service/UserService.java
package com.app.service;

import com.app.domain.user.User;
import com.app.dto.EmailDto;
import com.app.dto.UserDto;

public interface UserService {
    /**
     * 회원 등록
     * @param user
     * @return User
     */
    User registerUser(User user);
    /**
     * 회원 조회
     * @param email
     * @return User
     */
    User findByEmail(String email);
    /**
     * 회원 비밀번호 체크
     * @param rawPassword
     * @param encodedPassword
     * @return boolean
     */
    boolean checkPassword(String rawPassword, String encodedPassword);
    /**
     * 월시작일 업데이트
     * @param userDto
     * @return void
     */
    void updateMonthStartDate(UserDto userDto);
    /**
     * 비밀번호 변경
     * @param userDto
     * @return UserDto
     */
    UserDto updatePassword(User user, UserDto userDto);
    /**
     * 임시 비밀번호 이메일 전송
     * @param email
     * @return EmailDto
     */
    void sendPasswordResetEmail(String email);
    /**
     * 이메일 내용 설정
     * @param userDto
     * @return EmailDto
     */
    EmailDto setEmailContent(UserDto userDto);
}

File: frontend/src/service/user_service.js
import axios from '../util/axios'
import cookies from 'vue-cookies'

const API_URL = window.location.protocol + '//' + window.location.host + '/user/';

class UserService {
    /**
     * 로그인
     * @param {*} user 
     * @returns 
     */
    login(user) {
        return axios
            .post(API_URL + 'login', {
                email: user.email,
                password: user.password
            })
            .then(response => {
                if (response.data.token) {
                    cookies.set("user", response.data);
                }

                return response.data;
            });
    }
    /**
     * 로그아웃
     * @param {*}
     * @returns
     */
    logout() {
        cookies.remove("user");
    }
    /**
     * 테스트 계정 로그인
     * @returns
     */
    testIdLogin() {
        return axios
            .get(API_URL + 'login/test-id')
            .then(response => {
                if (response.data.token) {
                    cookies.set("user", response.data);
                }
                return response.data;
            });
    }
    /**
     * 회원등록
     * @param {*} userDto
     * @returns
     */
    register(userDto) {
        return axios.post(API_URL + 'register', userDto);
    }
    /**
     * 회원등록 확인
     * @param {*} email
     * @returns
     */
    selectRegisterCheck(email) {
        return axios
            .get(API_URL + 'register/check', { params: { email: email } })
            .then(response => {
                return response.data;
            });
    }
    /**
     * 임시 비밀번호 이메일 전송
     * @param {*} email
     * @returns
     */
    sendPasswordResetEmail(email) {
        return axios
            .get(API_URL + 'send/password-reset', { params: { email: email } })
            .then(response => {
                return response.data;
            });
    }
    /**
     * 월시작일 업데이트
     * @param {*} userDto
     * @returns
     */
    updateMonthStartDate(userDto) {
        return axios
            .post(API_URL + 'update/month-start-date', userDto)
            .then(response => {
                return response;
            });
    }
    /**
     * 비밀번호 변경
     * @param {*} userDto
     * @returns 
     */
    updatePassword(userDto) {
        return axios
            .post(API_URL + 'update/password', userDto)
            .then(response => {
                return response.data;
            });
    }


}

export default new UserService();